File: src/main/java/ch/uzh/ifi/access/model/dto/SubmissionDTO.java
package ch.uzh.ifi.access.model.dto;

import ch.uzh.ifi.access.model.constants.Command;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
public class SubmissionDTO {
    public boolean restricted = true;
    public String userId;
    public Command command;
    public List<SubmissionFileDTO> files = new ArrayList<>();
}

File: src/main/java/ch/uzh/ifi/access/model/dto/SubmissionDTO.kt
package ch.uzh.ifi.access.model.dto

import ch.uzh.ifi.access.model.constants.Command
import lombok.Data
import lombok.NoArgsConstructor

@Data
class SubmissionDTO(
    var restricted: Boolean = true,
    var userId: String? = null,
    var command: Command? = null,
    var files: List<SubmissionFileDTO> = ArrayList()
)